server/org.eclipse.emfcloud.ecore.glsp/src/main/java/org/eclipse/emfcloud/ecore/glsp/model/EcoreModelServerAccess.java

```diff
@@ -123,12 +136,19 @@ public boolean removeEReference(EcoreModelState modelState, EReference eReferenc
 			}
 			return this.editCompound(compoundCommand);
 		}
-		return this.remove(modelState, eReference, feature);
+		return this.edit(createRemoveEStructuralFeatureCommand(modelState, eReference));
+	}
+
+	private Command createRemoveEClassifierCommand(EcoreModelState modelState, EClassifier eClassifier) {
+		EditingDomain editingDomain = EcoreModelState.getEditorContext(modelState).getResourceManager()
+				.getEditingDomain();
+		int index = eClassifier.getEPackage().getEClassifiers().indexOf(eClassifier);
```

Comment:
I agree, it would feel more safe, but I stumbled upon a problem with that: On notifiying the subscribers, the command cannot be correctly encoded, as the value (i.e. eObject) is not stored inside the resource anymore. What do you think about sticking to the index for now?

Reply:
:+1: